core/src/edu/cornell/gdiac/main/CollisionController.java

```diff
@@ -224,13 +224,25 @@ private void setUpBounce(Contact contact, PlayerModel plyr, Obstacle bd1, Obstac
       if (bd1 instanceof BouncePlatformModel) {
         BouncePlatformModel bplt = (BouncePlatformModel) bd1;
         float c = bplt.getCoefficient();
+        boolean vertical=bplt.isVertical();
+        if (vertical&&(Math.abs(plyr.getPosition().x-bplt.getPosition().x)-(plyr.getWidth()+bplt.getWidth())/2f)+.1f<0){
```

Comment:
Two small things: leave a comment about what this math is doing since it is not very readable at first glance, and make sure to run your formatter which will also help with readability.